elyra/pipeline/registry.py

```diff
@@ -523,6 +523,20 @@ def parse_component_properties(self, component_body, component_path):
 
                     new_parameter_info = self.build_parameter(arg, class_name, class_content)
 
+                    component_parameters['parameters'].append({"id": new_parameter_info['parameter_ref']})
+                    if 'data' in new_parameter_info:
+                        if 'format' in new_parameter_info['data']:
+                            component_parameter_format = new_parameter_info['data']['format']
+                            if component_parameter_format:
+                                if component_parameter_format == 'str' or component_parameter_format == 'string':
+                                    if default_value is None:
+                                        default_value = ''
+                                elif component_parameter_format == 'int':
+                                    if default_value is None:
+                                        default_value = 0
+                                elif component_parameter_format == 'bool' or component_parameter_format == 'Boolean':
```

Comment:
We should cover `dict` and `list` as well. Have any ideas on how to cover other custom types?
I think @kevin-bates suggested another metadata layer in here so we can properly define them